File: sdk/python/spiderweb_protocol/examples/fs_hello.py
import asyncio

from spiderweb_protocol import FsClient, TextTransport


class DemoTransport(TextTransport):
    async def send_text(self, text: str) -> None:
        print("send:", text)

    async def receive_text(self) -> str:
        raise RuntimeError("provide a real websocket transport adapter")

    async def close(self) -> None:
        return None


async def main() -> None:
    client = FsClient(DemoTransport())
    await client.hello()


asyncio.run(main())

File: sdk/typescript/spiderweb-protocol/examples/fs-hello.ts
import {
  FsClient,
  WebSocketTextTransport,
} from "../src/index.js";

async function main(): Promise<void> {
  const transport = await WebSocketTextTransport.connect("ws://127.0.0.1:18891/fs");
  const client = new FsClient(transport);

  console.log(await client.hello(1));
  console.log(await client.lookup(2, 42, "README.md"));

  await transport.close();
}

void main();
